### PR TITLE
Improvements for password and Mastodon server input.

### LINKED ISSUE #21: mastodon base_url should be configurable
Took me way too long to realise that BASE_URL needed to be set in the Python script. It ought to be configurable, same as Twitter API keys and Mastodon log in creds.

I will probably try to patch this myself and submit a PR.

### PR DESCRIPTION
**Pull request by Spaxe**
This PR fixes #21. In addition, password should be input via `getpass.getpass()`. It is very terrible when you're typing password in plaintext, stored in terminal. This PR fixes that.

### CODE REVIEW

**Comment by Spaxe** [comment]
This PR also adds the choice of Mastodon server other than `mastodon.social` as part of the command prompt.

File: MastodonToTwitter.py [hunk 1]
```diff
@@ -39,7 +38,7 @@
 TWITTER_RETRY_DELAY = 20
 
 # Media regex, to trim media URLs
-MEDIA_REGEXP = re.compile(re.escape(MASTODON_BASE_URL.rstrip("/")) + "\/media\/(\w)+(\s|$)+")
+MASTODON_BASE_URL = ''
```
Comment (tobico) [inline MastodonToTwitter.py]: Is this declaration needed? Also, it looks like the comment is out of date

File: MastodonToTwitter.py [hunk 2]
```diff
@@ -114,8 +113,28 @@ def calc_expected_status_length(status, short_url_length = 23):
...
+
+        if MASTODON_BASE_URL == '':
+            # The Mastodon instance base URL. By default, https://mastodon.social/
+            MASTODON_BASE_URL = "https://mastodon.social"
+        elif not MASTODON_BASE_URL.startswith('https://'):
+            MASTODON_BASE_URL = 'https://' + MASTODON_BASE_URL
+
+        print("\n")
+        if os.path.isfile("mtt_mastodon_server.secret"):
+            print("You already have mastodon server set up, so we're skipping that step.")
```
Comment (tobico) [inline MastodonToTwitter.py]: "Mastodon" should start with a capital letter.

File: MastodonToTwitter.py [hunk 3]
```diff
@@ -114,8 +113,28 @@ def calc_expected_status_length(status, short_url_length = 23):
...
+        else:
+            print("Recording mastodon server...")
```
Comment (tobico) [inline MastodonToTwitter.py]: Same here

File: MastodonToTwitter.py (hunk 1 above)
Comment (Spaxe) [reply MastodonToTwitter.py]: I wasn't sure at first, so I left it in. Checking it now, it looks like it's declared separately on creating a new app, or loading from secrets. Removed.

**Comment by Spaxe** [comment]
Thanks @tobico, addressed your comments.

File: MastodonToTwitter.py [hunk 4]
```diff
@@ -114,8 +113,15 @@ def calc_expected_status_length(status, short_url_length = 23):
...
     while mastodon_works == False:
+        MASTODON_BASE_URL = input("Mastodon server (e.g. mastodon.social): ").strip()
         MASTODON_USERNAME = input("Mastodon Username (e-mail): ").strip()
-        MASTODON_PASSWORD = input("Mastodon Password: ").strip()
+        MASTODON_PASSWORD = getpass.getpass("Mastodon Password: ").strip()
+
+        if MASTODON_BASE_URL == '':
+            # The Mastodon instance base URL. By default, https://mastodon.social/
+            MASTODON_BASE_URL = "https://mastodon.social"
+        elif not MASTODON_BASE_URL.startswith('https://'):
```
Comment (halcy) [inline MastodonToTwitter.py]: If somebody enters something starting with http:// this breaks, I think.

File: MastodonToTwitter.py [hunk 5]
```diff
@@ -114,8 +113,15 @@ def calc_expected_status_length(status, short_url_length = 23):
 
     mastodon_works = False
     while mastodon_works == False:
+        MASTODON_BASE_URL = input("Mastodon server (e.g. mastodon.social): ").strip()
```
Comment (halcy) [inline MastodonToTwitter.py]: Should probably indicate that just pressing enter results in default.

**Review by halcy — CHANGES_REQUESTED** [review]
There's two things I'd like to be changed before I merge (or I can change them after), see comments

**Comment by Spaxe** [comment]
Thanks @halcy. I added `https://` in front of the prompt, so user will know not to type the protocol (and if they did, the connection will fail and make them redo it). Added "press Enter for" default also.

**Review by Spaxe — COMMENTED** [review]
Test comment.

**Comment by Spaxe** [comment]
Doesn't look like I can mark this as "Requested Changes Made", so it's up to you now. :)

**Comment by halcy** [comment]
I can contribute to open source projects while waiting in line at the post office. Modern technology is incredible.

Thanks for your work!

**Comment by Spaxe** [comment]
More to come - I love your work! Thanks for making transitioning to Mastodon easier.

**Pull request merged**